### DISCUSSION TITLE
Discussion about the Internationalisation effort for Zed

### DISCUSSION THREAD

**Discussion by ncs-pl** [zed-industries/zed#7461] [Feature Requests]
*This is a following of #7433 (related to ticket #7409).*

Hello Zed team,

While I do understand your point in the merge request's reply that changes should first be discussed, I would like to advocate for implementing an internationalisation system within the editor early in the development process since I believe it avoids scale issues that happen when implemented later, e.g. the need to update 100 strings v. update 100000k strings.

My main objective with the creation of this discussion is to spread the idea of i18n within the team, however, I can also recommend a few ideas to implement such as feature.
Indeed, unlike the [`rust_i18n`](https://lib.rs/crates/rust-i18n) recommended by Carlos Lopez (@foxkdev) in the ticket, I do encourage the adoption of Mozilla's [Project Fluent](https://projectfluent.org/), which is the result of various researches in the domain of translating software.

In hope the team reconsider the choice against early i18n support,

Nicolas Paul

**Comment by foxkdev** [comment]
Thanks to create Discussion about this,

Agree with priorize now vs after, now impact its better vs 1 month, or 1 year.

Im investigate Project Fluent, and i18n have pluralization feature, other features I think are not "better" to translators, In my experience, translators want .PO file or .json or platform to translate texts directly and not think in pluralization, genders etc

And in this app, I think only require pluralization, not others, But for me fluent are a valid solution too.

Agree to reconsider the early i18n support.

**Reply by ncs-pl** [reply]
Thanks for your reply Carlos!

Please tag me whenever plans for i18n starts if any help is needed (both translation and code implementation)!

**Reply by xnousnow** [reply]
Is this still being considered?

**Reply by imnotlxy** [reply]
Seems that weblate supports fluent

**Comment by chatterzhao** [comment]
Is this discussion still alive? I'm trying to implement i18n with extensions, and I think this solution is more flexible, and you can switch it on the fly instead of compiling it

**Reply by ice909** [reply]
> Is this discussion still alive? I'm trying to implement i18n with extensions, and I think this solution is more flexible, and you can switch it on the fly instead of compiling it

Does Zed currently support extensions similar to VS Code’s language packs (for example, installing a Simplified Chinese language pack to localize the UI)?

**Comment by Shuai-S** [comment]
Hello Zed team,
        Please ask if there has been progress in i18n, as more and more people around the world are trying to use Zed now, but the existing language barriers are hindering many from deepening their use, Is there consideration to accelerate the implementation of i18n?

**Comment by lvzhenbo** [comment]
Dear Zed Development Team,
With the release of the Windows version, Zed usage will significantly increase. Multilingual internationalization support should be prioritized.
I support providing i18n support through the plugin system, and an official multilingual plugin repository similar to VS Code should be established.

**Comment by getpool** [comment]
Multilingual internationalization support is urgently needed.

**Comment by openaddr** [comment]
Currently, it seems that the mainstream internationalization solution is implemented through plugins, and allowing users to switch the client language on the fly is a fundamental requirement.

**Comment by Super1Windcloud** [comment]
very agree , hope  consider  it!

**Comment by MrTact** [comment]
[Localization/i18n proposal](https://github.com/zed-industries/zed/discussions/43592)

**Comment by larkzhang** [comment]
When is the internationalization (i18n) feature expected to be launched? Many users are looking forward to localized UI support. Is there an official roadmap or an estimated release window?

**Comment by xx025** [comment]
期待

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
